atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/ConsensusForgettingStore.java

```diff
@@ -56,13 +55,14 @@ default void mark(Set<Cell> cells) {
      *   another update operation has failed in the past non-deterministically
      *   3. The semantics of the update operation depend on the underlying implementation.
      */
-    void atomicUpdate(Cell cell, byte[] value) throws KeyAlreadyExistsException;
+    AtomicUpdateResult atomicUpdate(Cell cell, byte[] value);
 
     /**
      * Performs atomic updates on multiple cells. This call may or may not guarantee atomicity across cells
      * depending on the underlying implementation.
+     * @return atomic update result for each individual cell
      */
-    void atomicUpdate(Map<Cell, byte[]> values) throws KeyAlreadyExistsException;
+    Map<Cell, AtomicUpdateResult> atomicUpdate(Map<Cell, byte[]> values);
```

Comment:
AtomicUpdateResult should be similar to KeyAlreadyExistsException, telling you about partial successes